presto-parser/src/main/java/com/facebook/presto/sql/tree/Analyze.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class Analyze
+        extends Statement
+{
+    private final Table table;
+    private final List<Property> properties;
+
+    public Analyze(Table table, List<Property> properties)
+    {
+        this(Optional.empty(), table, properties);
+    }
+
+    public Analyze(NodeLocation location, Table table, List<Property> properties)
+    {
+        this(Optional.of(location), table, properties);
+    }
+
+    private Analyze(Optional<NodeLocation> location, Table table, List<Property> properties)
+    {
+        super(location);
+        this.table = requireNonNull(table, "table is null");
+        this.properties = requireNonNull(properties, "properties is null");
+    }
+
+    public Table getTable()
+    {
+        return table;
+    }
+
+    public List<Property> getProperties()
+    {
+        return properties;
+    }
+
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context)
+    {
+        return visitor.visitAnalyze(this, context);
+    }
+
+    @Override
+    public List<? extends Node> getChildren()
+    {
+        return ImmutableList.<Node>builder()
+                .addAll(properties).build();
```

Comment:
- what about the `table`?
- move `.build()` to separate line

Reply:
Good catch. I missed that when changing from a `QualifiedName` member to a `Table` member.